### PR TITLE
Request dataplane operator access

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

**Review by andreer — APPROVED** [review]

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/athenz/AccessControlService.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+package com.yahoo.vespa.hosted.controller.api.integration.athenz;
+
+import com.yahoo.vespa.athenz.api.AthenzUser;
+
+import java.time.Instant;
+import java.util.Collection;
+
+public interface AccessControlService {
```
Comment (oyving) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/athenz/AccessControlService.java:10]: Javadoc? Call it `OperatorAccessControlService`?

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/athenz/AthenzAccessControlService.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        if (users.contains(user)) {
+            zmsClient.approvePendingRoleMembership(dataPlaneAccessRole, user, expiry);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Collection<AthenzUser> listMembers() {
+        return null;
```
Comment (oyving) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/athenz/AthenzAccessControlService.java:38]: At least throw `UnsupportedOperation`?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/ControllerApiHandler.java [hunk 3]
```diff
@@ -77,9 +90,47 @@ private HttpResponse get(HttpRequest request) {
...
+        if (path.matches("/controller/v1/access/requests/{user}")) return approveMembership(request, path.get("user"));
+        if (path.matches("/controller/v1/access/grants/{user}")) return grantAccess(request, path.get("user"));
         return notFound(path);
     }
 
+    private HttpResponse approveMembership(HttpRequest request, String user) {
+        AthenzUser athenzUser = AthenzUser.fromUserId(user);
+        byte[] jsonBytes = toJsonBytes(request.getData());
+        Inspector inspector = SlimeUtils.jsonToSlime(jsonBytes).get();
+        ApplicationId applicationId = ApplicationId.fromSerializedForm(inspector.field("applicationId").asString());
```
Comment (oyving) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/ControllerApiHandler.java:102]: Use SlimeUtils to throw if not exists?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/ControllerApiHandler.java [hunk 4]
```diff
@@ -77,9 +90,47 @@ private HttpResponse get(HttpRequest request) {
...
+        ZoneId zone = ZoneId.from(inspector.field("zone").asString());
```
Comment (oyving) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/controller/ControllerApiHandler.java:103]: SlimeUtils

**Review by oyving — APPROVED** [review]

**Review by oyving — APPROVED** [review]

**Pull request merged**